mungegithub/mungers/approvers/owners_test.go

```diff
@@ -669,3 +669,76 @@ func TestRemoveSubdirs(t *testing.T) {
 		}
 	}
 }
+
+func TestGetApprovalStatus(t *testing.T) {
+	tests := []struct {
+		testName       string
+		filenames      []string
+		approvers      []string
+		owners         map[string]sets.String
+		expectedStatus map[string]sets.String
+	}{
+		{
+			testName:       "Empty PR",
+			filenames:      []string{},
+			approvers:      []string{},
+			owners:         map[string]sets.String{},
+			expectedStatus: map[string]sets.String{},
+		},
+		{
+			testName:       "No approvers",
+			filenames:      []string{"a/a", "c"},
+			approvers:      []string{},
+			owners:         map[string]sets.String{"": sets.NewString("RootOwner")},
+			expectedStatus: map[string]sets.String{"": sets.String{}},
+		},
+		{
+			testName: "Approvers approves some",
+			filenames: []string{
+				"a/a",
+				"c/c",
+			},
+			approvers: []string{"CApprover"},
+			owners: map[string]sets.String{
+				"a": sets.NewString("AApprover"),
+				"c": sets.NewString("CApprover"),
+			},
+			expectedStatus: map[string]sets.String{
+				"a": sets.String{},
+				"c": sets.NewString("CApprover"),
+			},
+		},
+		{
+			testName: "Multiple approvers",
+			filenames: []string{
+				"a/a",
+				"c/c",
+			},
+			approvers: []string{"RootApprover", "CApprover"},
+			owners: map[string]sets.String{
+				"":  sets.NewString("RootApprover"),
+				"a": sets.NewString("AApprover"),
+				"c": sets.NewString("CApprover"),
+			},
+			expectedStatus: map[string]sets.String{
+				"a": sets.NewString("RootApprover"),
+				"c": sets.NewString("RootApprover", "CApprover"),
+			},
+		},
+		{
+			testName:       "Case-insensitive approvers",
```

Comment:
well, case-insensitive comparison doesn't really have a "direction". So if it matches different case, it's good to go.

I carefully selected that we should use the name/case from the approvers list rather than the name/case from the OWNERS files, but that could be more explicit. The reason is that OWNERS are manually written (hence can contain errors), while the approvers list is generated directly from your exact github handle.